src/main/java/seedu/address/logic/commands/FindRestaurantCommand.java

```diff
@@ -0,0 +1,68 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.model.Model;
+import seedu.address.model.restaurant.LocationContainsKeywordsPredicate;
+import seedu.address.model.restaurant.RNameContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all restaurants in restaurant book whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class FindRestaurantCommand extends Command {
+
+    public static final String COMMAND_WORD = "(rt)find";
+    public static final String COMMAND_FUNCTION = "Finds all restaurants whose names contain any of "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all restaurants whose names contain "
+            + "any of the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: " + COMMAND_WORD + " [r/RESTAURANT_NAME] [l/LOCATION] \n"
+            + "Example: " + COMMAND_WORD + " r/McDonalds ";
```

Comment:
I thought we can filter by other criteria as well like price point, cuisine etc.?

Reply:
And I think Joel used n/NAME for the restaurant name can prob use the same prefix?

Reply:
Ok! Agreed